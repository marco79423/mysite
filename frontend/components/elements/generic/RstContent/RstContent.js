import React from 'react'
import {Global, css, useTheme} from '@emotion/react'
import styled from '@emotion/styled'

import pygmentsStyle from './pygmentsStyle'

const Base = styled.div`
  margin-top: 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
`

const RstStyle = () => {
  const theme = useTheme()
  return (
    <Global styles={css`
      .rst-content {
        ${pygmentsStyle}
        hr {
          border: solid #dddddd;
          border-width: 1px 0 0;
          clear: both;
          margin: 1.25rem 0 1.1875rem;
          height: 0;
        }

        a, a:visited, a:hover {
          color: ${theme.global.link.color};
          text-decoration: none;
        }

        header {
          margin-bottom: 2rem;

          h1 {
            font-size: 2rem;
            margin-top: 0;
            margin-bottom: 0;
          }

          .meta {
            margin-top: 15px;

            .rubric {
              color: #d65b4b;
              font-weight: bold;
            }
          }
        }

        h2, h3, h4, h5 {
          color: ${theme.page.main.content.article.header.color};
        }

        h2 {
          font-size: 1.8rem;
        }

        h3 {
          font-size: 1.5rem;
        }

        h4 {
          font-size: 1.3rem;
        }

        h5 {
          font-size: 1.2rem;
        }

        ul {
          font-size: 1.2rem;
          margin-left: 1.2rem;

          li {
            list-style-type: circle;
            margin-left: 1.2rem;
          }
        }

        ol {
          font-size: 1.2rem;
          margin-left: 1.2rem;

          li {
            margin-left: 1.2rem;
          }
        }

        p {
          color: rgb(26, 26, 26);
          font-size: 1.2rem;
          line-height: 1.8rem;

          &.rubric {
            font-weight: bold;
            font-size: larger;
            color: #d65b4b;
          }
        }

        pre {
          font-size: 1.2rem;
          font-family: Consolas, Arial, Microsoft JhengHei;
          color: ${theme.page.main.content.article.block.color};
          background: ${theme.page.main.content.article.block.background};
          border: 1px solid ${theme.page.main.content.article.block.borderColor};
          padding: 0.7rem;
          margin-bottom: 1rem;
          white-space: pre-wrap;

          &.doctest-block {
            border: 0;
            color: ${theme.page.main.content.article.doctestBlock.color};
            background: ${theme.page.main.content.article.doctestBlock.background};
          }
        }

        .highlight pre {
          font-size: 1.2rem;
          font-family: Consolas, Arial;
          color: #505050;
          background-color: #ECECEC;
        }

        table.highlighttable {
          pre {
            font-size: 1.2rem;
            font-family: Consolas, Arial;
            color: #505050;
            background-color: #ECECEC;
            border: none;
            padding: 0;
            margin-bottom: 0;
          }
        }

        table {
          border-collapse: collapse;
          width: 100%;

          th {
            background: ${theme.page.main.content.article.table.header.background};
            color: white;
            padding: 0.7rem;
            font-size: 1.2rem;
          }

          tr:nth-of-type(odd) {
            background: #eee;
          }

          td {
            font-size: 1.2rem;
            padding: 0.7rem;
            border: 1px solid #ccc;
          }
        }

        img {
          margin-bottom: 1rem;

          box-shadow: 2px 2px 2px rgb(0 0 0 / 70%);
          border: solid 1px;

          @media (max-width: 400px) {
            width: 100%;
          }
        }

        .note {
          background-color: ${theme.page.main.content.article.note.background};
          padding: 0.1rem 0.7rem 0.7rem 0.7rem;
          margin-bottom: 1rem;
          border: 1px solid ${theme.page.main.content.article.note.borderColor};

          /*解決 note 的問題*/

          p.first {
            font-size: 1.2rem;
            font-weight: bold;
            color: ${theme.page.main.content.article.note.titleColor};
            margin-bottom: 0.1rem;
          }

          p.last {
            /*line-height: 1rem;*/
            margin-bottom: 0;
          }
        }

        .social-block {
          margin-bottom: 20px;

          .fb-share-button {
            bottom: 5px;
            margin-right: 5px;
          }
        }

        .series-block {
          background: #ECECEC;
          padding: 20px;
          margin-bottom: 20px;

          .title {
            font-weight: 700;
            margin-bottom: 30px;
          }

          div:not(:first-of-type) {
            margin-top: 10px;
          }
        }
      }
    `}/>
  )
}

function RstContent({content}) {
  return (
    <div>
      <RstStyle/>
      <Base className="rst-content" dangerouslySetInnerHTML={{__html: content}}/>
    </div>
  )
}

export default React.memo(RstContent)

