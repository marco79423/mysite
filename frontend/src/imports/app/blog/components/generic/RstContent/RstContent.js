import React from 'react'
import PropTypes from 'prop-types'
import styled, {injectGlobal, withTheme} from 'styled-components'

import pygmentsStyle from './pygmentsStyle'

const Base = styled.div`
  margin-top: 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
`

@withTheme
export default class RstContent extends React.PureComponent {
  static PropTypes = {
    theme: PropTypes.object.isRequired,
    content: PropTypes.string.isRequired
  }

  injectGlobalStyle = () => {
    injectGlobal`
      .rst-content {
        ${pygmentsStyle}
        
        hr {
          border: solid #dddddd;
          border-width: 1px 0 0;
          clear: both;
          margin: 1.25rem 0 1.1875rem;
          height: 0;
        }
      
        a, a:visited, a:hover {
          color: ${this.props.theme.global.link.color};
          text-decoration: none;
        }
      
        header {
          margin-bottom: 2rem;
      
          h1 {
            font-size: 2rem;
            margin-top: 0;
            margin-bottom: 0;
          }
      
          .meta {
            margin-top: 15px;
      
            .rubric {
              color: #d65b4b;
              font-weight: bold;
            }
          }
        }
      
        h2, h3, h4, h5 {
          color: ${this.props.theme.page.main.content.article.header.color};
        }
      
        h2 {
          font-size: 1.8rem;
        }
      
        h3 {
          font-size: 1.5rem;
        }
      
        h4 {
          font-size: 1.3rem;
        }
      
        h5 {
          font-size: 1.2rem;
        }
      
        ul {
          font-size: 1.2rem;
          margin-left: 1.2rem;
          li {
            margin-left: 1.2rem;
          }
        }
      
        ol {
          font-size: 1.2rem;
          margin-left: 1.2rem;
          li {
            margin-left: 1.2rem;
          }
        }
      
        p {
          color: rgb(26, 26, 26);
          font-size: 1.2rem;
          line-height: 1.8rem;
      
          &.rubric {
            font-weight: bold;
            font-size: larger;
            color: #d65b4b;
          }
        }
      
        pre {
          font-size: 1.2rem;
          font-family: Consolas, Arial, Microsoft JhengHei;
          color: ${this.props.theme.page.main.content.article.block.color};
          background: ${this.props.theme.page.main.content.article.block.background};
          border: 1px solid ${this.props.theme.page.main.content.article.block.borderColor};
          padding: 0.7rem;
          margin-bottom: 1rem;
          white-space: pre-wrap;
      
          &.doctest-block {
            border: 0;
            color: ${this.props.theme.page.main.content.article.doctestBlock.color};
            background: ${this.props.theme.page.main.content.article.doctestBlock.background};
          }
        }
      
        .highlight pre {
          font-size: 1.2rem;
          font-family: Consolas, Arial;
          color: #505050;
          background-color: #ECECEC;
        }
      
        table.highlighttable {
          pre {
            font-size: 1.2rem;
            font-family: Consolas, Arial;
            color: #505050;
            background-color: #ECECEC;
            border: none;
            padding: 0;
            margin-bottom: 0;
          }
        }
      
        table {
          border-collapse: collapse;
          width: 100%;
      
          th {
            background: ${this.props.theme.page.main.content.article.table.header.background};
            color: white;
            padding: 0.7rem;
            font-size: 1.2rem;
          }
      
          tr:nth-of-type(odd) {
            background: #eee;
          }
          td {
            font-size: 1.2rem;
            padding: 0.7rem;
            border: 1px solid #ccc;
          }
        }
      
        img {
          margin-bottom: 1rem;
      
          @media (max-width: 400px) {
            width: 100%;
          }
        }
      
        .note {
          background-color: ${this.props.theme.page.main.content.article.note.background};
          padding: 0.1rem 0.7rem 0.7rem 0.7rem;
          margin-bottom: 1rem;
          border: 1px solid ${this.props.theme.page.main.content.article.note.borderColor};
      
          /*解決 note 的問題*/
          p.first {
            font-size: 1.2rem;
            font-weight: bold;
            color: ${this.props.theme.page.main.content.article.note.titleColor};
            margin-bottom: 0.1rem;
          }
      
          p.last {
            /*line-height: 1rem;*/
            margin-bottom: 0;
          }
        }
      
        .social-block {
          margin-bottom: 20px;
      
          .fb-share-button {
            bottom: 5px;
            margin-right: 5px;
          }
        }
      
        .series-block {
          background: #ECECEC;
          padding: 20px;
          margin-bottom: 20px;
      
          .title {
            font-weight: 700;
            margin-bottom: 30px;
          }
      
          div:not(:first-child) {
            margin-top: 10px;
          }
        }
      }
    `
  }

  render() {
    this.injectGlobalStyle()
    return (
      <Base className='rst-content' dangerouslySetInnerHTML={{__html: this.props.content}}/>
    )
  }
}
