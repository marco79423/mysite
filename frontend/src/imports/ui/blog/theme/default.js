const colorTable = {
  'cyan-50': '#E0F7FA',
  'cyan-600': '#00ACC1',
  'cyan-700': '#0097A7',
  'cyan-800': '#00838F',
  'cyan-900': '#006064',
  'cyan-A700': '#00B8D4',

  'teal-50':'#E0F2F1',
  'teal-100':'#B2DFDB',
  'teal-200':'#80CBC4',
  'teal-300':'#4DB6AC',
  'teal-400':'#26A69A',
  'teal-500':'#009688',
  'teal-600':'#00897B',
  'teal-700':'#00796B',
  'teal-800':'#00695C',
  'teal-900':'#004D40',
  'teal-A700':'#00BFA5',

  'grey-50': '#FAFAFA',
  'grey-200': '#EEEEEE',
  'grey-300': '#E0E0E0',
  'grey-600': '#757575',
  'grey-700': '#616161',
  'grey-900': '#212121'
}

export default {
  global: {
    fontColor: colorTable['grey-900'],
    link: {
      color: colorTable['cyan-700'],
      hoverColor: colorTable['cyan-900']
    }
  },
  page: {
    background: colorTable['grey-700'],
    header: {
      background: colorTable['teal-400'],
      titleColor: colorTable['grey-50']
    },
    nav: {
      background: colorTable['teal-600'],
      menu: {
        color: '#ECECEC',
        background: colorTable['teal-600'],
        hoverBackground: colorTable['cyan-900'],
        dropdown: {
          background: colorTable['teal-600'],
          borderColor: colorTable['teal-800']
        }
      }
    },
    main: {
      background: colorTable['grey-200'],
      content: {
        article: {
          background: 'white',
          metadata: {
            tip: {
              color: '#FFFFFF',
              background: colorTable['teal-600']
            }
          },
          header: {
            color: colorTable['grey-700']
          },
          block: {
            color: 'white',
            background: colorTable['teal-600'],
            borderColor: '#ccc'
          },
          doctestBlock: {
            color: 'white',
            background: '#6D6D6D',
          },
          note: {
            background: '#ECECEC',
            titleColor: colorTable['teal-800'],
            borderColor: '#ccc'
          },
          table: {
            header: {
              background: colorTable['teal-600'],
            }
          }
        },
        pagination: {
          color: colorTable['grey-200'],
          background: colorTable['teal-700'],
          hoverBackground: colorTable['teal-900'],
          activeBackground: colorTable['teal-900'],
        }
      },
      aside: {
        background: colorTable['grey-200'],
        darkerBackground: colorTable['grey-600'],

        section: {
          titleColor: colorTable['grey-700'],
          borderColor: colorTable['grey-200']
        }
      }
    },
    footer: {
      background: colorTable['teal-600'],
      copyrightColor: '#ECECEC',
    }
  }
}