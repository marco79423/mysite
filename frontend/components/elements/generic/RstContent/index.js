import RstContent from './RstContent'

export default RstContent
