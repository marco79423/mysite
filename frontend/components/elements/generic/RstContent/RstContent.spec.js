import React from 'react'
import renderer from 'react-test-renderer'
import {ThemeProvider} from '@emotion/react'

import RstContent from './RstContent'
import theme from '../../../theme/default'

test('It should render RstContent component correctly', () => {
  const content = '<div>天氣不錯</div>'

  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <RstContent content={content}/>
    </ThemeProvider>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
