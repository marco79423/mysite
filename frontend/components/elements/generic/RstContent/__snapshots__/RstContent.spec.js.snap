// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render RstContent component correctly 1`] = `
<div>
  <div
    className="rst-content css-1htbla5 e1u79y390"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>天氣不錯</div>",
      }
    }
  />
</div>
`;
